repos/spack_repo/builtin/packages/py_numba/package.py

```diff
@@ -33,41 +43,50 @@ class PyNumba(PythonPackage):
-    depends_on("python@3.10:3.13", when="@0.61:", type=("build", "run"))
-    depends_on("python@3.9:3.12", when="@0.59:0.60", type=("build", "run"))
-    depends_on("python@3.8:3.11", when="@0.57:0.58", type=("build", "run"))
-    depends_on("python@3.7:3.10", when="@0.55:0.56", type=("build", "run"))
-    depends_on("python@3.7:3.9", when="@0.54", type=("build", "run"))
-
-    # max_numpy_run_version in setup.py is a non inclusive upper bound
-    # min_numpy_run_version < min_numpy_build_version and these ranges use
-    # min_numpy_build_version
-    depends_on("py-numpy@2.0:2.3", when="@0.62:", type=("build", "run"))
-    depends_on("py-numpy@2.0:2.2", when="@0.61.2", type=("build", "run"))
-    depends_on("py-numpy@2.0:2.1", when="@0.61.0", type=("build", "run"))
-    depends_on("py-numpy@2.0", when="@0.60", type=("build", "run"))
-    depends_on("py-numpy@1.22:1.26", when="@0.58.1:0.59", type=("build", "run"))
-    depends_on("py-numpy@1.21:1.25", when="@0.58.0", type=("build", "run"))
-    depends_on("py-numpy@1.21:1.24", when="@0.57", type=("build", "run"))
-    depends_on("py-numpy@1.18:1.23", when="@0.56.1:0.56.4", type=("build", "run"))
-    depends_on("py-numpy@1.18:1.22", when="@0.55.2:0.56.0", type=("build", "run"))
-    depends_on("py-numpy@1.18:1.21", when="@0.55.0:0.55.1", type=("build", "run"))
-    depends_on("py-numpy@1.17:1.20", when="@0.54", type=("build", "run"))
-    depends_on("py-setuptools", type=("build", "run"))
-    depends_on("py-llvmlite@0.45", when="@0.62", type=("build", "run"))
-    depends_on("py-llvmlite@0.44", when="@0.61", type=("build", "run"))
-    depends_on("py-llvmlite@0.43", when="@0.60", type=("build", "run"))
-    depends_on("py-llvmlite@0.42", when="@0.59", type=("build", "run"))
-    depends_on("py-llvmlite@0.41", when="@0.58", type=("build", "run"))
-    depends_on("py-llvmlite@0.40", when="@0.57", type=("build", "run"))
-    depends_on("py-llvmlite@0.39", when="@0.56", type=("build", "run"))
-    depends_on("py-llvmlite@0.38", when="@0.55", type=("build", "run"))
-    depends_on("py-llvmlite@0.37", when="@0.54", type=("build", "run"))
+    with default_args(type="build"):
+        depends_on("c")
+        depends_on("cxx")
+
+    with default_args(type=("build", "link", "run")):
+        # See min/max_*_version in setup.py
+        # Upper bounds are exclusive
+        depends_on("python@3.10:3.14", when="@0.63:")
+        depends_on("python@3.10:3.13", when="@0.61:0.62")
+        depends_on("python@3.9:3.12", when="@0.59:0.60")
+        depends_on("python@3.8:3.11", when="@0.57:0.58")
+        depends_on("python@3.7:3.10", when="@0.55:0.56")
+        depends_on("python@3.7:3.9", when="@0.54")
+
+    with default_args(type=("build", "run")):
+        # Use min_numpy_run_version, not min_numpy_build_version
+        # min_numpy_build_version may be higher to ensure backwards-compatibility of wheels,
+        # but this doesn't matter for Spack which always guarantees compatibility
```

Comment:
This is one change from the previous recipe. When building binaries for PyPI, you have to build with numpy 2+ to ensure compatibility with both numpy 1 and 2. However, we don't build generic ABI-agnostic binaries with Spack, so there's no reason to require numpy 2+.